test/e2e/observer/registry/observer/observer-commands.sh

```diff
@@ -7,22 +7,21 @@ set -o xtrace
 
 function cleanup() {
   echo "Running cleanup after being terminated."
-  echo -n "cancelled" > "${SHARED_DIR}/cancelled"
-  exit 0
 }
 
 trap cleanup EXIT
 trap cleanup INT
 
-while true; do
-    if [[ -f "${KUBECONFIG}" ]]; then
-      echo "\$KUBECONFIG exists"
-      break
-    fi
-    echo "\$KUBECONFIG does not exist, waiting..."
-    sleep 1
+echo "do-not-upload-me" >"${SHARED_DIR}/intruder"
+
+for (( i=1; i<=300; i++ )); do 
```

Comment:
Is it really needed to wait 300s?

Reply:
I don't think so, it's just an hard limit. After a lot of run I noticed kubelet can take up to ~2mins before propagating the changes from the secret to the pods.